src/Codeception/Test/Descriptor.php

```diff
@@ -53,18 +53,28 @@ public static function getTestSignatureUnique(\PHPUnit\Framework\SelfDescribing
     public static function getTestAsString(\PHPUnit\Framework\SelfDescribing $testCase)
     {
         if ($testCase instanceof \PHPUnit\Framework\TestCase) {
-            $text = $testCase->getName();
-            $text = preg_replace('/([A-Z]+)([A-Z][a-z])/', '\\1 \\2', $text);
-            $text = preg_replace('/([a-z\d])([A-Z])/', '\\1 \\2', $text);
-            $text = preg_replace('/^test /', '', $text);
-            $text = ucfirst(strtolower($text));
-            $text = str_replace(['::', 'with data set'], [':', '|'], $text);
+            $text = self::getTestCaseNameAsString($testCase->getName());
```

Comment:
It would be better to avoid modifying this method, it is used in a few places and could cause unwanted effects.
If you really need different output, please create a different method, but I think that it can be avoided by using named datasets in dataprovider.

Reply:
Please see https://github.com/Codeception/Codeception/pull/5690/commits/3754bb018299f7f757c1779323df64f767907ccb

* I did not modify the behaviour of the method, but just extracted part of it into a 2nd method for re-use to avoid duplication of the production code, as due to changing the test case methods (better names as you suggested) the expectation had to be adapted as well (see usage of `getTestCaseNameAsString()` in test case) and required the specially written test case name.
* IIRC named datasets in dataprovider also does not work (anymore) as documented; as a workaround there is e.g. overridden `__toString()` in https://github.com/Codeception/Codeception/pull/5690/commits/dde81b809d99333504e49fb3c146c5cbb2e915ad

By taking small steps/changes at a time and several commits I had hoped to ease your reviewing process...

Reply:
A problem with commits last time was that the first line of all commit messages was the same, so I didn't bother looking at individual commits.

Put the most important information in the first line.

Reply:
OK, sorry, I'll try that in the future